### ISSUE TITLE
A question about generating grid networks

### ISSUE THREAD

**Issue by amiradavoudi** [eclipse-sumo/sumo#13941]
Hello

Could you please help me determine if it is possible to generate different grid networks with randomly varying connectivity among the junctions, as well as different directions between the junctions?

For example, in one grid network, there may be no connectivity between junctions A and B, but in another grid network, such a connection might exist. Similarly, in one grid network, the direction between junctions C and D may be one-way, while in another grid, it is bidirectional.

In summary, I would like to generate various types of grid networks with random connectivity and random directions, whether they are one-way or two-way, within the generated network.

Thank you in advance for your support.

**Comment by angelobanse** [comment]
Use [netgenerate](https://sumo.dlr.de/docs/netgenerate.html) to generate the desired grid network, and use the option **--bidi-probability** with a value less than 1 (1 means all connections have a reverse edge).

E.g. `netgenerate --grid --grid.number 4 --bidi-probability 0.8 -o example.net.xml`

You can also use netgenerate from within netedit: File > Generate Network.
In the resulting dialog select "Grid", specify a location where to store the resulting network and click on "Advanced" to set the **--bidi-probability** value.

**Comment by amiradavoudi** [comment]
Thank you for the response. Could you also elaborate on the connectivity between each of the two nodes, as mentioned in the question? I mean that there is a possibility that we have no connection between two nodes in a grid; for example, between nodes A and B there is no edge.

**Comment by palvarezlopez** [comment]
@amiradavoudi you can also generate the random network with Netedit (file/generate network), and after edit the network manually, removing the non neccesary edges.

**Comment by amiradavoudi** [comment]
@palvarezlopez  I need to generate lots of grid networks so removing the non-necessary edges takes a lot of time, which is not feasible for our project. There is no way to do it using a specific command in SUMO? Should I write a script by myself to change the generated XML file?

**Comment by angelobanse** [comment]
You can define an edgeType file (`example.typ.xml`) with two dummy edgeTypes:

```
<type>
    <type id="myType_0"/>
    <type id="myType_1"/>
</type>
```

And load that file in the netgenerate dialog (look under "input" for the "type-files" option). You must also enable the "random-type" checkbox.

After the network is created, go to the Select mode ![](https://sumo.dlr.de/docs/images/icon_modeselect.png) (in netedit) and select all edges with one of those dummy types, and then delete them.

![grafik](https://github.com/eclipse-sumo/sumo/assets/16121932/e76eeb27-62d2-4bb3-b25e-ecb691155c93)
